### PR TITLE
Adds weights calculator for unbalanced classes

### PR DESCRIPTION
**Pull request by brenoarosa**
Hi.  
This PR contains a function similar to scikit-learn's  [compute_sample_weight](http://scikit-learn.org/dev/modules/generated/sklearn.utils.class_weight.compute_sample_weight.html).  

The implementation differs from sklearn in the signature as **n_classes** is required since unique op is missing [[issue]](https://github.com/pytorch/pytorch/issues/2031) and if **class_weight** are not provided it is computed to balance class influence.
The output may differ by a normalization factor too.  

I know that the usual process would be to create a issue before pull requesting but I'm sending it as I already had the code. Feel free to reject if this don't fit.

### CODE REVIEW

**Comment by weiyangfb** [comment]
If we want this, maybe it should be placed in/closed to `sampler.py`. Thoughts? cc @soumith @apaszke
